runtime/doc/autocmd.txt

```diff
@@ -729,7 +729,22 @@ MenuPopup			Just before showing the popup menu (under the
 					o	Operator-pending
 					i	Insert
 					c	Command line
-							*OptionSet*
+							*ModeChanged*
+ModeChanged			After changing the mode. The pattern is
+				matched against `'old_mode:new_mode'`, for
+				example match against `i:*` to simulate
+				|InsertLeave|.
+				The following values of |v:event| are set:
+					old_mode The mode before it changed.
+					new_mode The new mode as also returned
+					         by |mode()|.
+				When ModeChanged is triggered, old_mode will
+				have the value of new_mode when the event was
+				last triggered.
+				Usage example to use relative line numbers
+				when entering visual mode: >
+					:autocmd ModeChanged *:v set rnu
```

Comment:
In case you find out your original Vim patch was wrong, this should be `*:v*` instead of `*:v`.